ludwig/data/dataframe/pandas.py

```diff
@@ -63,3 +66,97 @@ def partitioned(self):
 
 
 PANDAS = PandasEngine()
+
+
+# Pandas to TFRecord adapter.
+# The code below is modified based on:
+# https://github.com/schipiga/pandas-tfrecords/blob/master/pandas_tfrecords/to_tfrecords.py
+def pandas_df_to_tfrecords(df,
+                           path,
+                           compression_type='GZIP',
+                           compression_level=9,
+                           columns=None):
+    schema = get_schema(df, columns)
+    tfr_iters = get_tfrecords(df, schema)
+    write_tfrecords(tfr_iters, path,
+                    compression_type=compression_type,
+                    compression_level=compression_level)
+
+
+def get_schema(df, columns=None):
```

Comment:
This should be fine for now. One thing we can improve in the future is to use Ludwig's `training_set_metadata` to more precisely obtain the data type of each column. But if this works, this is fine for now.